pkg/analysis/passes/checksum/checksum_test.go

```diff
@@ -0,0 +1,277 @@
+		interceptor.Diagnostics[0].Title,
+	)
+}
+
+func TestSkipsCheckSumWhenNoProvided(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+
+	pass := &analysis.Pass{
+		ResultOf: map[*analysis.Analyzer]interface{}{},
+		CheckParams: &analysis.CheckParams{
+			Checksum:              "", // no checksum
+			ArchiveCalculatedMD5:  "f9b1c42c45cbf4953d7da5c31b3d73d9",
+			ArchiveCalculatedSHA1: "187cd80948f240957bf8399745335b89f005c5f0",
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	_, err := Analyzer.Run(pass)
+	assert.NoError(t, err)
+
+	// should not call the report function
+	require.Len(t, interceptor.Diagnostics, 0)
+}
+
+func TestValidMD5ChecksumFromUrl(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+
+	pass := &analysis.Pass{
+		ResultOf: map[*analysis.Analyzer]interface{}{},
+		CheckParams: &analysis.CheckParams{
+			Checksum:              "https://example.com/checksum",
+			ArchiveCalculatedMD5:  "f9b1c42c45cbf4953d7da5c31b3d73d9",
+			ArchiveCalculatedSHA1: "187cd80948f240957bf8399745335b89f005c5f0",
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterResponder("GET", "https://example.com/checksum",
+		httpmock.NewStringResponder(200, "f9b1c42c45cbf4953d7da5c31b3d73d9 "))
+
+	_, err := Analyzer.Run(pass)
+	assert.NoError(t, err)
+
+	require.Len(t, interceptor.Diagnostics, 0)
+}
+
+func TestValidMD5ChecksumFromUrlWithSpaces(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+
+	pass := &analysis.Pass{
+		ResultOf: map[*analysis.Analyzer]interface{}{},
+		CheckParams: &analysis.CheckParams{
+			Checksum:              "https://example.com/checksum",
+			ArchiveCalculatedMD5:  "f9b1c42c45cbf4953d7da5c31b3d73d9",
+			ArchiveCalculatedSHA1: "187cd80948f240957bf8399745335b89f005c5f0",
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterResponder("GET", "https://example.com/checksum",
+		httpmock.NewStringResponder(200, "f9b1c42c45cbf4953d7da5c31b3d73d9 archive-file-name.zip"))
+
+	_, err := Analyzer.Run(pass)
+	assert.NoError(t, err)
+
+	require.Len(t, interceptor.Diagnostics, 0)
+}
+
+func TestInvalidMD5ChecksumFromUrl(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+
+	pass := &analysis.Pass{
+		ResultOf: map[*analysis.Analyzer]interface{}{},
+		CheckParams: &analysis.CheckParams{
+			Checksum:              "https://example.com/checksum",
+			ArchiveCalculatedMD5:  "f9b1c42c45cbf4953d7da5c31b3d73d9",
+			ArchiveCalculatedSHA1: "187cd80948f240957bf8399745335b89f005c5f0",
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterResponder("GET", "https://example.com/checksum",
+		httpmock.NewStringResponder(200, "x9b1c42c45cbf4953d7da5c31b3d73d3"))
```

Comment:
```suggestion
	httpmock.RegisterResponder("GET", "https://example.com/checksum",
	httpmock.NewStringResponder(200, "x9b1c42c45cbf4953d7da5c31b3d73d3"))
```

Reply:
this is not an indentation error but intended, this is a function parameter split in two lines so the indentation exists to show that they are part of the same line. (this is auto-formatted by https://github.com/segmentio/golines)

Reply:
My bad :facepalm: